Project/DAQ_System/Program/Src/Sensor/GyroScope/l3gd20h.hpp

```diff
@@ -0,0 +1,105 @@
+        L3GD20H(I2C_HandleTypeDef &hi2c);
+        virtual ~L3GD20H() = default;
+
+        /// Converts the raw values from the gyroscope to Degrees/second
+        /// @return ± 250 on each axis
+        virtual short* GetDegreesPerSecond() override;
+        /// Converts the raw values from the gyroscope to Radians/second
+        /// @return ± 4.363 on each axis
+        virtual float* GetRadiansPerSecond() override;
+
+    private:
+        /// takes in raw data from the gyroscope
+        /// @param g[3] array to hold data for each axis
+        /// @return -32768 to 32767 on each axis
+        bool ReadRawData(short g[3]);
+
+        /// takes in raw data from the gyroscope
+        /// @return -32768 to 32767 on each axis
+        virtual bool ComputeAngularVelocity() override;
+
+        /// writes to a specified value to a specified register
+        /// @param addr_i2c 8-bit slave address( 7-bit slave address << 1 )
+        /// @param addr_reg 8-bit address of register to write to
+        /// @param register_data 8-bit value to write to selected register
+        /// @return write status (true or false)
+        bool WriteReg(uint8_t addr_i2c, uint8_t addr_reg, uint8_t register_data);
+
+
+        /// reads values from the selected register
+        /// @param  addr_i2c 8-bit slave address( 7-bit slave address << 1 )
+        /// @param  addr_reg 8-bit address of register to write to
+        /// @param  read_data is array for storing the read content of the selected register
+        /// @return read status (true or false)
+        bool ReadReg(uint8_t addr_i2c,uint8_t *addr_reg, uint8_t read_data[]);
+
+
+        /// sequential reading function for l3g20h. Reads all the registers that contain axis
+        /// data. [0x28, 0x29, 0x2A, 0x2B, 0x2C, 0x2D]
+        /// @param  data_buf array to contain all the raw axis data from l3gd20h
+        /// @return status of function(true or false)
+        bool Receive(uint8_t data_buf[]);
+
+        /// averages the stationary values for each axis
+        /// @return None
+        void ComputeInitialOffset();
+
+        // initialization register for CTRL1 register initializes x,y, and z axis and PD to 1
+        uint8_t ctrl_reg1_init_value = 0x0F;
+
+        // FSR value given by the data sheet. The values is used to normalize the raw data
+        // to ± 250 degrees
+        static constexpr float DPS_250_Sensitivity = .00875f;
+
+        //Conversion factor of (2pi/360) to convert to radians
+        static constexpr float DegreesToRadians = 0.01745329251f; //Conversion facotr of (2pi/360)
+
+        // First element is the x-axis
+        // Second element is the y-axis
+        // Third element is the z-axis
+        short degrees_per_second_[3] = {0};
+        float radians_per_second_[3] = {0};
+        short angular_velocity_[3] = {0};
+        short offset_average_[3] = {0};
+
+        /// The I2C peripheral from ST's HAL library.
+        I2C_HandleTypeDef& hi2c_;
```

Comment:
I like to remove the "h" to make it more distinct from HAL's `hi2c` object. To me, the "h" kind of stands for HAL but it's also a practice in C to call an object/pointer/interface a "handle".

Let me know what you think! Perhaps something to add to our style guide.

Reply:
I like the idea of removing the "h" but at the same time, I think the "h" might be useful for newer members and those who aren't familiar with the HAL. As you said, it's a good practice to have the "h" to dictate we are using a handle. So keeping it can serve as a good reference for other members to learn good practices as they move on to make their own libraries and such.  in the end, it's honestly up to you, I don't mind either way.